driver/src/orderbook/streamed/state.rs

```diff
@@ -186,6 +212,61 @@ impl State {
         balance.balance += event.burnt_fees;
         Ok(())
     }
+
+    fn apply_trade(
+        &mut self,
+        user: UserId,
+        order: OrderId,
+        executed_sell_amount: u128,
+        executed_buy_amount: u128,
+        batch_id: BatchId,
+        revert: bool,
```

Comment:
I'm not a fan of passing this boolean around, but I guess otherwise we would have to duplicate a ton of code? 

Alternatively we could have one method that gets a combined result of all dependencies (buyToken, sellToken, balances, order) and then do the +/- in the functions directly? 

But since it's an implementation detail it's also not a big deal